pkg/apis/monitoring/v1/types.go

```diff
@@ -1816,22 +1816,34 @@ type PrometheusRuleSpec struct {
 // or 'abort'.  More info: https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md#partial-response
 // +k8s:openapi-gen=true
 type RuleGroup struct {
-	Name                    string `json:"name"`
-	Interval                string `json:"interval,omitempty"`
-	Rules                   []Rule `json:"rules"`
+	// Name of the rule group.
```

Comment:
I have added it here, adding it on `Name` was not working since it needs list type. I have added an e2e test for checking this as well

```
// +k8s:openapi-gen=true
type PrometheusRuleSpec struct {
	// Content of Prometheus rule file
	// +listType=map
	// +listMapKey=name
	Groups []RuleGroup `json:"groups,omitempty"`
}
```

Reply:
sorry my bad :)